### PR TITLE
resolve 14-fix-ts-lint-and-class-components

### PR DESCRIPTION
**Pull request by g00gol**

### CODE REVIEW

**Comment by g00gol** [comment]
ill try to convert class-based components to functional components in this mr

File: client/src/pages/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+import { PlanMarkerData } from "../dataObjects/PlanMarkerData";
+import { HomeProps } from "../types/HomeTypes";
+import {
+  radius as DEFAULT_RADIUS,
+  EPlaces,
+  keys,
+} from "../constants/GoogleMaps/config";
+import nearbySearch from "../api/GoogleMaps/nearbySearch";
+import SearchResults from "../components/Home/SearchResults";
+import TripWindow from "../components/Home/TripWindow";
```
Comment (g00gol) [inline client/src/pages/Home.tsx:29]: created two components `components/Home/SearchResults.tsx` and `components/Home/TripWindow.tsx` to house respective concerns

File: client/src/pages/Home.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+import { CircleData } from "../dataObjects/CircleData";
+import { PlaceData } from "../dataObjects/PlaceData";
+import { PlanMarkerData } from "../dataObjects/PlanMarkerData";
+import { HomeProps } from "../types/HomeTypes";
+import {
+  radius as DEFAULT_RADIUS,
+  EPlaces,
+  keys,
+} from "../constants/GoogleMaps/config";
+import nearbySearch from "../api/GoogleMaps/nearbySearch";
```
Comment (g00gol) [inline client/src/pages/Home.tsx:27]: outsourced nearbySearch logic to `api/GoogleMaps/nearbySearch.tsx`

File: client/src/pages/Home.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import { TripContext } from "../contexts/TripContext";
+import { CircleData } from "../dataObjects/CircleData";
+import { PlaceData } from "../dataObjects/PlaceData";
+import { PlanMarkerData } from "../dataObjects/PlanMarkerData";
+import { HomeProps } from "../types/HomeTypes";
+import {
+  radius as DEFAULT_RADIUS,
+  EPlaces,
+  keys,
+} from "../constants/GoogleMaps/config";
```
Comment (g00gol) [inline client/src/pages/Home.tsx:26]: made config file for google maps constants

File: client/src/main.tsx [hunk 4]
```diff
@@ -1,24 +1,21 @@
...
   <React.StrictMode>
     <BrowserRouter>
-      <AuthProvider>
-        <TripProvider>
-          <App />
-        </TripProvider>
-      </AuthProvider>
+      <ContextProvider>
+        <App />
+      </ContextProvider>
```
Comment (g00gol) [inline client/src/main.tsx:18]: new `<ContextProvider />` component removes the need to keep nesting our context providers. instead, see `contexts/Context.tsx`

File: client/src/components/PlaceCard.tsx [hunk 5]
```diff
@@ -1,167 +1,169 @@
...
+
+  const date: number = new Date().getDay();
+
+  const { addPlace, removePlace } = useContext(TripContext);
+
+  const showDetailsHandler = () => {
+    setShowDetails(true);
+    const map = mapRef.current;
+    if (!map || !place.place_id) {
+      console.log("Map or place ID is undefined");
```
Comment (bigguyhere) [inline client/src/components/PlaceCard.tsx:29]: Should we have some sort of error handling here besides just a console printout?

File: client/src/pages/Home.tsx [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    <div className="flex h-screen flex-col">
+      {props.children /* modal */}
+      <Navbar />
+      {isLoaded ? (
+        <div className="flex flex-grow">
+          <GoogleMap
+            mapContainerStyle={{ width: "100vw" }}
+            center={centerData}
+            zoom={15}
+            id={"TEST_MAP"}
```
[the remark this answers is not in the document]
Comment (g00gol) [reply client/src/pages/Home.tsx:114]: do we even need the id? we can remove it if u want

File: client/src/types/PlaceCardType.tsx
[the remark this answers is not in the document]
Comment (bigguyhere) [reply client/src/types/PlaceCardType.tsx:1]: Actually I believe this file is still being used, but the other three I do not believe are.

File: client/src/pages/Home.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (g00gol) [reply client/src/pages/Home.tsx:111]: i see it being used in App.css

**Pull request merged**